packages/push/src/PushRegistration.ts

```diff
@@ -24,249 +22,38 @@ export interface PushRegistrationOptions {
-            "categories": categories
-          };
-
-          return this.httpClient.post(PushRegistration.API_PATH, postData)
-          .then(() => {
-              if (isCordovaAndroid() && this.variantId) {
-                this.subscribeToFirebaseTopic(this.variantId);
-                if (options.categories) {
-                  for (const category of options.categories) {
-                    this.subscribeToFirebaseTopic(category);
-                  }
-                }
-              }
-
-              const storage = window.localStorage;
-              storage.setItem(PushRegistration.REGISTRATION_DATA_KEY, JSON.stringify(postData));
-
-              this.push
-              .on("notification", (notification: any) => {
-                if (PushRegistration.onMessageReceivedCallback) {
-                  PushRegistration.onMessageReceivedCallback(notification);
-                }
-              });
-
-              resolve();
-            }
-          )
-          .catch(reject);
-
-        } else {
-          // It should never happend but...
-          return reject(new Error("Push is not properly configured"));
-        }
-
-      });
-    });
+  public async register(options: PushRegistrationOptions | PushRegistrationWebpushOptions = {}): Promise<void> {
+    await this.delegate.register(options);
   }
 
   /**
-   * Unregister device for Android or IOS platforms
+   * Unregister the application from the UPS.
    */
-  public unregister(): Promise<void> {
-
-    if (this.validationError) {
-      return Promise.reject(new Error(this.validationError));
-    }
-
-    const storage = window.localStorage;
-    const jsonCachedData = storage.getItem(PushRegistration.REGISTRATION_DATA_KEY);
-
-    let postData;
-    let deviceToken = "";
-    let categories: string[] = [];
-
-    if (jsonCachedData) {
-      postData = JSON.parse(jsonCachedData);
-      deviceToken = postData.deviceToken;
-      categories = postData.categories;
-    }
-
-    if (!deviceToken) {
-      return Promise.reject(new Error("Device token should not be empty"));
-    }
-
-    return new Promise((resolve, reject) => {
-      if (this.httpClient) {
-        const endpoint = PushRegistration.API_PATH + "/" + deviceToken;
-        return this.httpClient.delete(endpoint, {})
-        .then(() => {
-
-          if (isCordovaAndroid() && this.variantId) {
-            this.unsubscribeToFirebaseTopic(this.variantId);
-            if (categories) {
-              for (const category of categories) {
-                this.unsubscribeToFirebaseTopic(category);
-              }
-            }
-          }
-
-          storage.removeItem(PushRegistration.REGISTRATION_DATA_KEY);
-
-          resolve();
-        })
-        .catch(reject);
-      } else {
-        // It should never happend but...
-        return reject(new Error("Push is not properly configured"));
-      }
-    });
+  public async unregister(): Promise<void> {
+    this.delegate.unregister();
```

Comment:
Do we need to add an `await` here like in `register()`?

Reply:
👍